src/azure-cli/azure/cli/command_modules/resource/_packing_engine.py

```diff
@@ -94,8 +115,7 @@ def _pack_artifacts(cmd, template_abs_file_path, context):
             TemplateSpecTemplateArtifact = get_sdk(cmd.cli_ctx, ResourceType.MGMT_RESOURCE_TEMPLATESPECS,
                                                    'TemplateSpecTemplateArtifact', mod='models')
             template_content = read_file_content(abs_local_path)
-            sanitized_template = _remove_comments_from_json(template_content)
-            template_json = json.loads(json.dumps(sanitized_template))
+            template_json = json.loads(json.dumps(process_template(template_content)))
```

Comment:
> json.loads(json.dumps(

Why serialize and then deserialize the object from `process_template()` ? The data returned by `process_template` has been deserialized~

Reply:
data returned by process_template is a ordered dictionary due to  shell_safe_json_parse, the template_json needs to be a json object.

Reply:
OK